src/p11_child/p11_child_openssl.c

```diff
@@ -381,8 +383,23 @@ static errno_t do_ocsp(struct p11_ctx *p11_ctx, X509 *cert)
 
     OCSP_request_add1_nonce(ocsp_req, NULL, -1);
 
-    ocsp_resp = process_responder(ocsp_req, host, path, port, use_ssl,
-                                  req_timeout);
+    if (p11_ctx->timeout != -1  && p11_ctx->cert_verify_opts->soft_ocsp) {
+        /* decrease timeout of time so far spend in p11_child */
+        /* substract 1 so we finish before child is forcibly terminated */
+        req_timeout = p11_ctx->timeout - 1 -
+            (time(NULL) - p11_ctx->start_timestamp);
+        if (req_timeout < 0) {
```

Comment:
Not needed if below condition changed to `if (req_timeout <= 0)`

Reply:
Only to have a dedicated debug message.
Otherwise instead of two:
```
if (req_timeout <= 0) {
if (req_timeout != 0) {
```
there could be single
```
if (req_timeout > 0) { process_responder() }
```

Reply:
rewritten a bit to make it more readable. Note that -1 has special meaning - unspecified timeout.